Display entire Scratch stage on the Sense Hat with one block

ericrosenbaum opened the pull request:
It's fun to use the Sense Hat extension to display letters, drawings, and individual pixels. But what if you could directly display the contents of the Scratch stage on the RGB Matrix, with a single block? 

I thought that would be cool, because then you could make colorful animations on the matrix using the existing tools in scratch: 
 
![pi-matrix-rainbow2-sm](https://user-images.githubusercontent.com/567844/68893868-0db56500-06f4-11ea-8b84-877352291c37.gif)

And! you can even make games that you play with the sense hat joystick (just make your sprites really big!):

![pi-matrix-pong2](https://user-images.githubusercontent.com/567844/68893875-1148ec00-06f4-11ea-9efd-a5ab943a2064.gif)

So I've added a single block, "display entire stage," that takes the currently displayed scratch stage (including any sprites, backdrops, and pen drawings that are visible), crops and scales it down to an 8x8 pixel image, and sends it to the RGB matrix. If you put this into a forever loop, and drag a sprite around on the stage, or animate it with blocks, you see it moving on the matrix.

This is just a crazy idea but I'm curious what folks think! Some open questions include:

- What should the block be called? Currently it is "display entire stage", but that isn't exactly right, because the stage is cropped to a square in order to display it on the matrix. "display stage" is good but doesn't seem _awesome_ enough. 
- Is it okay to put this block first?
- Is the performance okay? (I haven't noticed any issues)
- If you open up Scratch and click this block with the default cat showing, you can only just barely see a couple orange pixels. I wonder if there's a way to make that first experience with this block better.

martinohanlon commented:
Thanks a lot @ericrosenbaum. Very Exciting. Simon pointed this out to me and asked me to gather some thoughts and opinions. I havent been able to do, but am hopefully I will be able to pick it up next week.

martinohanlon commented:
@ericrosenbaum sorry for taking a little while to get back to you. I finally found some time to talk a look at this.

In summary, I think it opens up opportunities for those new scratch and the sense hat to do something exciting quickly and that it extends the options for experienced scratchers to create solutions which would be impractical with the existing blocks.

Some specifics though:
- it dont think it should be the "top block", it shouldn't distract from the existing display blocks which have (arguably) more learning value
- I think the name should be simplified to `display stage`
- I share your concerns that when using this block with the default stage and sprite, you dont see much
- Because of this ^ what do you think about also including block which would `display sprite` - my thoughts are that this block would scale the sprite to fill the 8x8 display. This would give a much clearer visual response.
- performance seemed really good. tbh, I was really surprised how quickly the stage rendered.

ericrosenbaum commented:
thanks @martinohanlon! I think the `display sprite` block is a great idea- I'll investigate.

ericrosenbaum commented:
@martinohanlon okay, ready for re-review!

- added "display sprite" block
- renamed the other block to "display stage"
- re-ordered the blocks

here's a little demo of "display sprite" showing star, strawberry, earth and scratch cat:
![sensehat-sprites](https://user-images.githubusercontent.com/567844/70458239-b6e74500-1a7f-11ea-9739-69dc74f53e58.gif)

martinohanlon commented:
Thanks a lot @ericrosenbaum 

I got chance to test it out today. I really like the display sprite addition, I think it makes it much simpler for scratchers to understand what is happening and gives the display stage block more context.

@spl237 When you think its best, I'm happy for this to be merged into the extension.
